internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -29,13 +30,25 @@ var (
 )
 
 // GetTlsConfigurationForClient returns a tls.Config instance for use by a client application.
-func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AService_SetUpSessionClient) (*tls.Config, error) {
+func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AService_SetUpSessionClient, tokenManager tokenmanager.AccessTokenManager, localIdentity *commonpbv1.Identity) (*tls.Config, error) {
+	var token string
```

Comment:
Could we extract the logic for how to produce the AuthenticationMechanism array into a helper function that can be used in both GetTlsConfigurationForClient and GetTlsConfigurationForServer?

We can model it on the v1 code here: https://github.com/google/s2a-go/blob/8249b7935631c5db8ad4d77e1d15a965e14c7d48/internal/handshaker/handshaker.go#L397